check-rbl-lists

```diff
@@ -0,0 +1,714 @@
+
+        It returns the found addresses as a list of ipaddress objects.
+        """
+
+        addresses = []
+
+        def request_alarm_caller(signum, sigframe):
+            raise RequestTimeoutError(self.timeout, host)
+
+        if use_timeout:
+            signal.signal(signal.SIGALRM, request_alarm_caller)
+            signal.alarm(self.timeout)
+
+        try:
+            for addrinfo in socket.getaddrinfo(host, 80):
+                if addrinfo[0] == socket.AF_INET or not only_ipv4:
+                    addr = self.to_ipaddress(addrinfo[4][0])
+                    addresses.append(addr)
+        except socket.gaierror as e:
+            raise HostNotFoundError(host, str(e))
+
+        if use_timeout:
+            signal.alarm(0)
+
+        if addresses and self.verbose > 2:
+            LOG.debug("Got addresses for host {!r}:\n".format(host) + pp(addresses))
+        return addresses
+
+    # -------------------------------------------------------------------------
+    def init_logging(self):
+        """
+        Initialize the logger object.
+        It creates a loghandler with all output to STDERR.
+
+        @return: None
+        """
+
+        log_level = logging.WARN
+        if self.verbose:
+            if self.verbose > 1:
+                log_level = logging.DEBUG
+            else:
+                log_level = logging.INFO
+
+        root_logger = logging.getLogger()
+        root_logger.setLevel(log_level)
+
+        # create formatter
+        format_str = ''
+        if self.verbose:
+            format_str = '[%(asctime)s]: '
+        format_str += self.appname + ': '
+        if self.verbose:
+            if self.verbose > 1:
+                format_str += '%(name)s(%(lineno)d) %(funcName)s() '
+            else:
+                format_str += '%(name)s '
+        format_str += '%(levelname)s - %(message)s'
+        formatter = logging.Formatter(format_str)
+
+        # create log handler for console output
+        lh_console = logging.StreamHandler(sys.stderr)
+        lh_console.setLevel(log_level)
+        lh_console.setFormatter(formatter)
+
+        root_logger.addHandler(lh_console)
+
+        return
+
+    # -------------------------------------------------------------------------
+    def __call__(self):
+        """Magic method to make the object callable."""
+        self.run()
+
+    # -------------------------------------------------------------------------
+    def run(self):
+        """Execute the underlaying actions."""
+        blacklists = copy.copy(self.blacklists)
+
+        cur_len = len(blacklists)
+        while cur_len > 1:
+            index = random.randrange(0, cur_len)
+            rbl = blacklists.pop(index)
+            addr_rec = self.ip_reverse + '.' + rbl
+            self.records2check.append(addr_rec)
+            cur_len = len(blacklists)
+
+        rbl = blacklists.pop(0)
+        addr_rec = self.ip_reverse + '.' + rbl
+        self.records2check.append(addr_rec)
```

Comment:
What is the reason behind that randomise and the usage of an array `.pop`? Would an `for` loop not be simpler, faster and less code? Or do i miss something as non python expert?

```suggestion
        for rbl in blacklists:
            addr_rec = self.ip_reverse + '.' + rbl
            self.records2check.append(addr_rec)
```